cli/src/main/java/com/devonfw/tools/ide/io/FileAccessImpl.java

```diff
@@ -659,8 +659,22 @@ public static String generatePermissionString(int permissions) {
   private void extractArchive(Path file, Path targetDir, Function<InputStream, ArchiveInputStream> unpacker) {
 
     this.context.trace("Unpacking archive {} to {}", file, targetDir);
+    // calculate the size of archive for the progressbar max size
+    long sizeOfArchive = 0;
     try (InputStream is = Files.newInputStream(file); ArchiveInputStream ais = unpacker.apply(is)) {
       ArchiveEntry entry = ais.getNextEntry();
+      while (entry != null) {
+        sizeOfArchive += entry.getSize();
+        entry = ais.getNextEntry();
+      }
```

Comment:
did you test the overhead cause by this?
Does the read all data, unpack everything or will this just read some header/meta data and skip the rest of the payload?
As it seems the InputStream variants are slow in this regard.
We might need to implement this based on `ZipFile` instead:
https://stackoverflow.com/a/29331429/1506881

Reply:
Also if you do that workaround later for `TarArchiveEntry`, then IMHO the same has to be done here as well.

Reply:
Finally, there is not need for calculating the size of the archives. It's enough to take the file size. The issue before was probably that the ZIP extraction didn't work correctly and by increasing stepSize of the probgressbar the maximal size wasn't reached. Further, by using `ZipFile` we have the compressed size of entries and an overload iof the progressbar size doesn't occur anymore. 

Related issues: #575 and  #609